crates/diesel_models/src/query/fraud_check.rs

```diff
@@ -0,0 +1,54 @@
+use diesel::{associations::HasTable, BoolExpressionMethods, ExpressionMethods};
+use router_env::tracing::{self, instrument};
+
+use crate::{
+    errors, fraud_check::*, query::generics, schema::fraud_check::dsl, PgPooledConn, StorageResult,
+};
+
+impl FraudCheckNew {
+    #[instrument(skip(conn))]
+    pub async fn insert(self, conn: &PgPooledConn) -> StorageResult<FraudCheck> {
+        generics::generic_insert(conn, self).await
+    }
+}
+
+impl FraudCheck {
+    #[instrument(skip(conn))]
+    pub async fn update_with_attempt_id(
+        self,
+        conn: &PgPooledConn,
+        fraud_check: FraudCheckUpdate,
+    ) -> StorageResult<Self> {
+        match generics::generic_update_with_unique_predicate_get_result::<
+            <Self as HasTable>::Table,
+            _,
+            _,
+            _,
+        >(
+            conn,
+            dsl::attempt_id
+                .eq(self.attempt_id.to_owned())
+                .and(dsl::merchant_id.eq(self.merchant_id.to_owned())),
+            FraudCheckUpdateInternal::from(fraud_check),
+        )
+        .await
+        {
+            Err(error) => match error.current_context() {
+                errors::DatabaseError::NoFieldsToUpdate => Ok(self),
+                _ => Err(error),
+            },
+            result => result,
+        }
+    }
+
+    pub async fn get_with_payment_id(
```

Comment:
we can only use the indexed fields in where clause of the queries. 

We only have merchant_id <> payment_id in index and add merchant_id in this query.